.github/workflows/main.yml

```diff
@@ -29,107 +30,87 @@ jobs:
 
       - name: set commit to output
         id: getcommit
-        run: | 
+        run: |
            prsha=$(echo $response | jq '.[-1].sha'  | tr -d '"')
-           echo "::set-output name=sha::$prsha" 
-        env: 
+           echo "::set-output name=sha::$prsha"
+        env:
           response:  ${{ steps.get_PR_commits.outputs.data }}
-          
+
       - uses: actions/checkout@v2
         with:
-          ref: ${{steps.getcommit.outputs.sha}}      
+          ref: ${{steps.getcommit.outputs.sha}}
 
-      - name: Build docker image
+      - name: Cloning the litmus-e2e Repo
+        if: startsWith(github.event.comment.body, '/run-e2e-login') || startsWith(github.event.comment.body, '/run-e2e-workflow') || startsWith(github.event.comment.body, '/run-e2e-all')
```

Comment:
We can remove if conditions for cloning litmus, Installing Dependencies and Starting the server if these jobs are common for all task. It will run with jobs if condition that is it will run when it will see a command starting with /run-e2e

Reply:
Okay i will remove them, I thought we may add more actions which may not be related Cypress.

Reply:
Yes, If the jobs are not common then if can add that.